### PR TITLE
Make libraries build with Native AOT corelib

### PR DESCRIPTION
**Pull request by MichalStrehovsky**

### CODE REVIEW

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs [hunk 1]
```diff
@@ -1,61 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (jkotas) [inline src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs:1]: Should we keep this, and instead move `StackFrameExtensions.cs` to CoreLib and exclude it for CoreRT?

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Environment.CoreRT.Win32.cs [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
 using System.Runtime.InteropServices;
 
 namespace System
 {
     public static partial class Environment
     {
+#pragma warning disable CS8763 // A method marked [DoesNotReturn] should not return.
```
Comment (jkotas) [inline src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Environment.CoreRT.Win32.cs:11]: Is this needed when ExitProcess is marked DoesNotReturn? Sounds like a bug.

File: src/libraries/System.Runtime/src/MatchingRefApiCompatBaseline.txt [hunk 3]
```diff
@@ -17,4 +17,21 @@ MembersMustExist : Member 'public System.Runtime.Serialization.DeserializationTo
 MembersMustExist : Member 'public void System.Runtime.Serialization.SerializationInfo.ThrowIfDeserializationInProgress()' does not exist in the reference but it does exist in the implementation.
```
Comment (jkotas) [inline src/libraries/System.Runtime/src/MatchingRefApiCompatBaseline.txt:17]: These baselines may deserve another entry in https://github.com/dotnet/runtimelab/issues/52

**Comment by MichalStrehovsky** [comment]
```
'F:\workspace\_work\1\s\artifacts\obj\coreclr\nativeaot\System.Private.CoreLib\..\..\Windows_NT.x64.Debug\src\nativeaot\Runtime\Full\AsmOffsets.cs' could not be found. [F:\workspace\_work\1\s\src\coreclr\src\nativeaot\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

Of course it cannot, because the runtime was built into "Windows_NT.x64.Checked", but libs apparently don't know what "checked" is.

@jkotas What do you think - should we kill off the Checked builds in the CI in lieu of Debug? Debug is plenty fast and I never enjoyed debugging optimized code anyway (I don't remember if I ever built CoreCLR as checked either).

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs (hunk 1 above)
Comment (jkotas) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs:1]: The NativeAOT copy has different implementation of some of the methods, to allow you to get IPs for offline-processing even when the textual symbols are not available. IIRC, this was used by some of crash telemetry libraries for UWP.

These APIs are no-ops in the current shared version.

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Environment.CoreRT.Win32.cs (hunk 2 above)
Comment (jkotas) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Environment.CoreRT.Win32.cs:11]: Ok, it is weird that it works everywhere else. Wasn't it just an oversight?

File: src/libraries/System.Runtime/src/MatchingRefApiCompatBaseline.txt (hunk 3 above)
Comment (jkotas) [reply src/libraries/System.Runtime/src/MatchingRefApiCompatBaseline.txt:17]: ok

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs (hunk 1 above)
Comment (MichalStrehovsky) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs:1]: Ah, I see. So we should make S.D.StackTrace typeforward StackFrameExtensions to CoreLib when building with ClrNativeAot = true.

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Environment.CoreRT.Win32.cs (hunk 2 above)
Comment (MichalStrehovsky) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Environment.CoreRT.Win32.cs:11]: I'll try to submit it without this suppression, but I'm pretty sure I was getting build failure without this suppression and the DllImport is annotated.

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs (hunk 1 above)
Comment (jkotas) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs]: I would hope that we can concentrate all these differences in CoreLib, and the rest of the libraries can be identical.

For StackFrameExtensions, it means moving StackFrameExtensions.cs to System.Private.CoreLib for both JIT and NativeAOT.

**Pull request merged**